### ISSUE TITLE
What is the intended build process given your use of a private fork of docker-java?

### ISSUE THREAD

**Issue by fraz3alpha** [jenkinsci/docker-plugin#414]
Since commit ac2797a3966aac90152b0d78a75bf04c60469ec1 and the use of version `2.2.4-jenkins` of the docker-java dependency the plugin will not build as-is.

What is the intended build process now that you are using a non-standard dependency -do we have to build that library ourselves first and ensure it is in our local maven cache?

**Comment by KostyaSha** [comment]
That how @magnayn maintain, he solved his problem by using custom patched dependency. His "custom" change is in docker-java PR but he can't provide requested information. 
If you want to build custom hpi you can freely switch to mainstream docker-java version as i sure that you don't need his custom fix.
Have a nice day!

**Comment by magnayn** [comment]
Well, the process was going to be that I submitted the fix to docker-java and it wouldn't be the need.

Sadly I did, but was met with the response that "2.x branch is no longer supported" (despite the fact that I was - literally - supporting it by submitting a PR (!) 

Moving to 3.x required a bunch of, somewhat arbitary IMO, API changes that I didn't have time for + figuring out how to fix the same thing in docker-java 3.x (this free software is justified and funded by it's use in my $dayjob).

We now _are_ on the 3.x branch, as work needed some of the auth functionality.

I've once again submit PRs to cover the missing / broken parts; I'd clearly like to be on mainline. If they don't get merged, I'll add a fork as a submodule to keep things in one place.
